ocf_datapipes/select/trim_dates_with_insufficent_data.py

```diff
@@ -0,0 +1,86 @@
+"""
+This is a class function that slices a contigous datetime range to the 12th hour
+"""
+import logging
+
+import numpy as np
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+logger = logging.getLogger(__name__)
+
+
+@functional_datapipe("trim_dates_with_insufficent_data")
+class TrimDatesWithInsufficentDataIterDataPipe(IterDataPipe):
+    """Trim the date values of the Xarray Timeseries data"""
+
+    def __init__(self, source_datapipe: IterDataPipe, intervals: int):
+        """Trim the dates to the exact 12th hour
+
+        For the five minute interval, If the time_utc dates are insufficent and
+        less than multiple of 289, this method trims that extended dates
+
+        For example:
+            <xr.DataArray> time_utc : "2020-01-01T00:00"..."2020-01-02T00:00"..."2020-01-02T06:00"
+
+        This method trims the inusfficent less than one day data at
+        the end and provides full set of complete one day interval (5min or 15min or...)
+
+        Args:
+            source_datapipe: Xarray emitting timeseries data
+            intervals: Intervals respective of the time range
+                5min xarray interval data = 288
+                15min xarray interval data = 96
+                .........
+        """
+        self.source_datapipe = source_datapipe
+        self.intervals = intervals
```

Comment:
should this be `minimum_number_data_points`, `intervals` means something else to me

Reply:
yes, `minimum_number_data_points` is more sensible than `intervals` and easy to understand, making changes asap!